[fxa-oauth-server][train-141]: Failing test `POST /key-data fails with bad assertion`

AssertionError: expected 500 to equal 401

See https://circleci.com/gh/mozilla/fxa/44167?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link.

Up until now there was one test that always reported "pending" in the OAuth server, I suspect this was it. I'm not sure how this test ever passed.

The problem is no browserid-verifier runs on Circle. This test does not [mock out a verifier response](https://github.com/mozilla/fxa/blob/1f3c23fc0dbc6556b3b793b9a947113356f462c4/packages/fxa-auth-server/fxa-oauth-server/test/api.js#L3159), so an attempt is made at calling a real browserid verifier, which fails because the verifier cannot be reached. The 500 error is propagated back.

This can be seen locally by starting all of the servers, then killing all the auth servers, the oauth-server, and the browserid-verifier. Once all those are stopped, run the oauth tests. Only this one fails.

This is preventing the train-141 tests from passing and a new docker image from being built.

--- a/packages/fxa-auth-server/fxa-oauth-server/test/api.js
+++ b/packages/fxa-auth-server/fxa-oauth-server/test/api.js
@@ -3156,6 +3156,7 @@ describe('/v1', function() {
       });
 
       it('fails with bad assertion', () => {
+        mockAssertion().reply(200, VERIFY_FAILURE);
         return Server.api.post(genericRequest).then(res => {
           assert.equal(res.statusCode, 401);
           assertSecurityHeaders(res);

